cpp/ql/lib/experimental/semmle/code/cpp/dataflow/ProductFlow.qll

```diff
@@ -352,32 +355,119 @@ module ProductFlow {
         pragma[only_bind_out](succ.getNode().getEnclosingCallable())
     }
 
+    private newtype TKind =
+      TInto(DataFlowCall call) {
+        intoImpl1(_, _, call) or
+        intoImpl2(_, _, call)
+      } or
+      TOutOf(DataFlowCall call) {
+        outImpl1(_, _, call) or
+        outImpl2(_, _, call)
```

Comment:
Looks fine now. If you want, you can change these two disjunctions into conjunctions for a little bit of earlier filtering - might as well once it's written this way.

Reply:
Oh, yes... that's true. I guess I won't do that 😂.

Reply:
Double-reversal: Conjunctions will work if you change the negations in the jump-identification from using `into1` etc. to `intoImpl1` etc.